### DISCUSSION TITLE
Suggestion on read and write app user config file?

### DISCUSSION THREAD

**Discussion by KiddoV** [zauberzeug/nicegui#2605] [Q&A] [answered]
Every app needs a user config file. Is there any good config parser to read (JSON, ...) that suits NiceGUI?

Thanks,

**Comment by rodja** [comment]
NiceGUI uses orjson if possible because its much faster than the builtin json module.

**Reply by KiddoV** [reply]
Thanks for the suggestion!

A JSON library is quite... general. Are there any libraries specifically designed for handling configuration, similar to how Golang has libraries like [gookit/config](https://github.com/gookit/config) or the more comprehensive [spf13/viper](https://github.com/spf13/viper)?

**Comment by rodja** [comment, marked as answer]
We mostly use `.env` files for simple configuration parameters: https://pypi.org/project/python-dotenv/
Everything else is general json. But you can use what ever there is available in Python. NiceGUI has no objections 😉
